python/cudf/cudf/_lib/io/datasource.pxd

```diff
@@ -1,10 +1,11 @@
 # Copyright (c) 2020, NVIDIA CORPORATION.
 
-from libcpp.memory cimport unique_ptr
-
-from cudf._lib.cpp.io.types cimport datasource
+from cudf._lib.cpp.io.types cimport arrow_io_source, datasource
 
 
 cdef class Datasource:
-
     cdef datasource* get_datasource(self) nogil except *
+
+cdef class NativeFileDatasource(Datasource):
+    cdef arrow_io_source c_datasource
```

Comment:
If we decide not to have a default constructor for `arrow_io_source`, make this a `shared_ptr[arrow_io_source]` instead (`from libcpp.memory cimport shared_ptr`).

```suggestion
    cdef shared_ptr[arrow_io_source] c_datasource
```